lib/console.rb

```diff
@@ -0,0 +1,36 @@
+class Console
+
+  def initialize(output = $stdout, input = $stdin, move_validator)
+    @output = output
+    @input = input
+    @move_validator = move_validator
+  end
+
+  def present(message)
+    @output.puts message
+  end
+
+  def receive
+    @input.gets.chomp
+  end
+
+  def validated_input
```

Comment:
I like that you have separated a `Display` and a `Console` class, what do you think about separating a `Validator` class too? Should `Console` know about validation?